src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectInfo.cs

```diff
@@ -618,7 +618,8 @@ internal Type[] GetMemberTypes(string[] inMemberNames, Type objectType)
             bool isFound = false;
             for (int i = 0; i < _cache._memberInfos.Length; i++)
             {
-                if (!memberMissing && inMemberNames[i].Equals(_cache._memberInfos[i].Name))
+                if (!memberMissing && 
+                    inMemberNames[i].Equals(FormatterServices.GetFullMemberInfoName(_cache._memberInfos[i], objectType)))
```

Comment:
Could we instead bring back SerializationInfoField?  It would need some small tweaks, since RuntimeFieldInfo isn't public, so it would work with FieldInfo instead, but it looks like most everything else would "just work"?  By masking this in BinaryFormatter, we're fixing at least the places in BinaryFormatter we found, but a) there could be others, and b) FormatterServices.GetSerializableMembers returns these member infos, and so other code could be accessing the returned Name and expecting it to be the full name, so anyone else accessing it would need to apply a similar workaround as well.